flowmachine/flowmachine/features/subscriber/mds_volume.py

```diff
@@ -0,0 +1,113 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# -*- coding: utf-8 -*-
+"""
+Class for calculating MDS volume statistics.
+"""
+
+import warnings
+
+from ..utilities.sets import EventsTablesUnion
+from .metaclasses import SubscriberFeature
+
+valid_stats = {"count", "sum", "avg", "max", "min", "median", "stddev", "variance"}
+
+
+class MDSVolume(SubscriberFeature):
+    """
+    This class calculates statistics associated with MDS volume usage.
+
+    Parameters
+    ----------
+    start, stop : str
+         iso-format start and stop datetimes
+    volume: {"total", "upload", "download"}, default "total"
+        The type of volume.
+    statistic : {'count', 'sum', 'avg', 'max', 'min', 'median', 'mode', 'stddev', 'variance'}, default 'sum'
+        Defaults to sum, aggregation statistic over the durations.
+    hours : 2-tuple of floats, default 'all'
+        Restrict the analysis to only a certain set
+        of hours within each day.
+    subscriber_identifier : {'msisdn', 'imei'}, default 'msisdn'
+        Either msisdn, or imei, the column that identifies the subscriber.
+    subscriber_subset : str, list, flowmachine.core.Query, flowmachine.core.Table, default None
+        If provided, string or list of string which are msisdn or imeis to limit
+        results to; or, a query or table which has a column with a name matching
+        subscriber_identifier (typically, msisdn), to limit results to.
+    tables : str or list of strings, default 'events.mds'
```

Comment:
Does it make sense to have this param, given it explicitly only deals with MDS?

Reply:
I think this was for consistency sake. You never know if someone decides to create a curated events table with mds data.